src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+import * as React from 'react';
+import CodeMirror from 'codemirror';
+import clsx from 'clsx';
+import { useNexusContext } from '@bbp/react-nexus';
+import { useSelector } from 'react-redux';
+import {
+  TEditorPopoverResolvedData,
+  editorLinkResolutionHandler,
+  getTokenAndPosAt,
+  mayBeResolvableLink,
+} from './editorUtils';
+import { TDELink } from '../../store/reducers/data-explorer';
+import { RootState } from '../../store/reducers';
+import useResolutionActions from './useResolutionActions';
+
+const downloadImg = require('../../images/DownloadingLoop.svg');
+
+export const CODEMIRROR_HOVER_CLASS = 'CodeMirror-hover-tooltip';
+export const CODEMIRROR_LINK_CLASS = 'fusion-resource-link';
+type TTooltipCreator = Pick<
+  TEditorPopoverResolvedData,
+  'error' | 'resolvedAs' | 'results'
+>;
+
+function removePopoversFromDOM() {
+  const popovers = document.querySelectorAll(
+    `.${CODEMIRROR_HOVER_CLASS}-popover`
+  );
+  popovers.forEach(popover => popover.remove());
+}
+function removeTooltipsFromDOM() {
+  const tooltips = document.getElementsByClassName(CODEMIRROR_HOVER_CLASS);
+  tooltips &&
+    Array.from(tooltips).forEach(tooltip => {
+      tooltip.remove();
+    });
+}
+
+function createTooltipNode({
+  tag,
+  title,
+  isDownloadable,
+}: {
+  tag: string | null;
+  title: string;
+  isDownloadable?: boolean;
+}) {
+  const tooltipItemContent = document.createElement('div');
+  tooltipItemContent.className = 'CodeMirror-hover-tooltip-item';
+  const nodeTag = document.createElement('div');
+  nodeTag.className = 'tag';
+  tag && nodeTag.appendChild(document.createTextNode(tag));
+  tooltipItemContent.appendChild(nodeTag);
+
+  const nodeTitle = document.createElement('span');
+  nodeTitle.className = 'title';
+  nodeTitle.appendChild(document.createTextNode(title));
+  tooltipItemContent.appendChild(nodeTitle);
+
+  const nodeDownload = isDownloadable && document.createElement('img');
```

Comment:
Lines 60 to 63 can be wrapped inside a `if (isDownloadable)` block to avoid the repetition of `nodeDownload &&`